runtime/include/j9cfg.h.ftl

```diff
@@ -33,6 +33,10 @@ extern "C" {
 #endif
 
 #include "omrcfg.h"
+#include "vendor_version.h"
+#ifndef JVM_VERSION_STRING
```

Comment:
This doesn't seem extensible, if more defines are added in the future or something else needs to be overridden. I'm thinking vendor_version.h should just be included after openj9_version_info.h so it can override the values.

Reply:
`JVM_VERSION_STRING` is not defined within `OpenJ9 vendor_version.h` but it is defined in vendor specific (such as IBM) `vendor_version.h`.
So if `JVM_VERSION_STRING` is not defined, `openj9_version_info.h` will be included to supply `OpenJ9` specific defines. Here checking `JVM_VERSION_STRING` is defined is to avoid checking `OPENJ9_BUILD` which requires adding the flag in `j9cfg.h` references.  
In the future if more defines are added, they are need to be provided within `vendor_version.h` (for vendor) or `OpenJ9 vendor_version.h` (for `OpenJ9`).
Will update the comments.

Reply:
I don't understand the argument. Imagine if there are two variables which are defined in openj9_version_info.h. How would you write the #if? i.e. #if '!defined(A) && !defined(B)` or '!defined(A) || !defined(B)`. What happens if vendor_version.h only needs to define one of them?

Reply:
Ideally the includes in `j9cfg.h` should look like following:
```
#if defined(OPENJ9_BUILD)
#include "../util/openj9_version_info.h"
#endif 
```
However currently `OPENJ9_BUILD` flag is enabled for individual `c/cpp` files via `module.xml` and not for head files. It can be changed to move the flag definition to the top makefile level. I feel it should be done in a separated PR. 
At mean time, this PR proposes an intermediate fix to remove the check for `JVM_VERSION_STRING` and include `openj9_version_info.h` anyway. Since this head file is only generated for `OpenJ9`, so vendor specific builds (such as IBM SDK) have to add such a head file but empty.

@DanHeidinga @pshipton does this make sense?

Reply:
It makes sense to me to define `OPENJ9_BUILD` at the top level for all files, instead of for individual files. Still vendor_version.h needs to be included after openj9_version_info.h

Reply:
Updated PR to include `vendor_version.h` after `openj9_version_info.h`.